flex-templates/java/regional_dlp_transform/src/main/java/com/google/cloud/blueprints/datawarehouse/BigQueryDynamicWriteTransform.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.blueprints.datawarehouse;
+
+import com.google.api.services.bigquery.model.TableCell;
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.auto.value.AutoValue;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryUtils;
+import org.apache.beam.sdk.io.gcp.bigquery.DynamicDestinations;
+import org.apache.beam.sdk.io.gcp.bigquery.TableDestination;
+import org.apache.beam.sdk.io.gcp.bigquery.WriteResult;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.ValueInSingleWindow;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@AutoValue
+@SuppressWarnings("serial")
+public abstract class BigQueryDynamicWriteTransform
+    extends PTransform<PCollection<KV<String, TableRow>>, WriteResult> {
+  public static final Logger LOG = LoggerFactory.getLogger(BigQueryDynamicWriteTransform.class);
+
+  public abstract String projectId();
+
+  public abstract String datasetId();
+
+  public static Builder newBuilder() {
+    return new AutoValue_BigQueryDynamicWriteTransform.Builder();
+  }
+
+  private static final String jsonSchema = "{"
```

Comment:
how is this schema const used in the code? was it for debug/test?

Reply:
Test purpose. It will be removed